x/nexus/keeper/wasm_message_route.go

```diff
@@ -35,6 +35,10 @@ func NewMessageRoute(nexus types.Nexus, account types.AccountKeeper, wasm types.
 			return err
 		}
 
+		if err := nexus.SetMessageExecuted(ctx, msg.ID); err != nil {
```

Comment:
Why not set failed if err is not nil?

Reply:
I find it to be bad practice in case there's an side effect/state transition within the method that should have reverted. We could use a RunCached but not as efficient as just letting it revert (which we do for routing to other modules as well)

Reply:
(if you mean, set it as failed if SetMessageExecuted returns an error, that would allow the message to be retried)